src/components/20-molecules/datepicker/utils/date.js

```diff
@@ -16,55 +18,74 @@ const getWeekdays = (date, locale) => {
   return out;
 };
 
-const ALL_DATE_SEPERATORS = / |,|\.|-|\//;
+const ALL_DATE_SEPARATORS = / |,|\.|-|\//;
+
 const clearStringFromIEGeneratedCharacters = string =>
   string.replace(/[^\x00-\x7F]/g, ''); // eslint-disable-line no-control-regex
+
+const zeroFill = (number, width) => {
+  const _number = number | 0; // coerce number to integer
+  const n_ = Math.abs(_number);
+  const zeros = Math.max(0, width - Math.floor(n_).toString().length);
+  const zeroString = (10 ** zeros).toString().slice(1);
+  return `${_number < 0 ? '-' : ''}${zeroString}${_number}`;
+};
```

Comment:
Understanding what this method does is a pain in the ass. What does `zeroFill` even mean? I would add a more declarative method name.

Reply:
Fair point. I could have added the stackoverflow URL...

Instead I refactored it and named it `addLeadingZeroes`.